solver/src/ethereum.rs

```diff
@@ -0,0 +1,20 @@
+use crate::settlement::Settlement;
+use anyhow::Result;
+use async_trait::async_trait;
+use model::TokenPair;
+use primitive_types::H160;
+
+// Not final at all. Just want some prototypes available to use in other code.
+
+#[async_trait]
+pub trait SettlementContract {
+    async fn get_nonce(&self, token_pair: TokenPair) -> Result<u32>;
+    async fn settle(&self, settlement: Settlement) -> Result<()>;
+}
+
+#[async_trait]
+pub trait Ethereum {
```

Comment:
one of the very difficult tasks will be to decide whether the driver should just query regularly an etherum-node about the erc20 balances and allowances or whether we need to subscribe to events about erc20 updates.

This decision is important for sure, no matter which solution nonce solution we will implement. Maybe it makes sense to investigate the best solution for this, assuming we have 20, 100, 300 orders per batch, before writing nonce-related code. Just and idea :)